gcp/lib/src/metadata.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+
+import 'bad_configuration_exception.dart';
+
+enum MetadataValue {
+  /// A set of typical environment variables that are likely to represent the
+  /// current Google Cloud instance region.
+  ///
+  /// For context, see:
+  /// * https://cloud.google.com/functions/docs/env-var
+  /// * https://cloud.google.com/compute/docs/gcloud-compute#default_project
+  /// * https://github.com/GoogleContainerTools/gcp-auth-webhook/blob/08136ca171fe5713cc70ef822c911fbd3a1707f5/server.go#L38-L44
+  ///
+  /// Note: these are ordered starting from the most current/canonical to least.
+  /// (At least as could be determined at the time of writing.)
+  project(
+    path: 'project/project-id',
+    environmentValues: {
+      'GCP_PROJECT',
+      'GCLOUD_PROJECT',
+      'CLOUDSDK_CORE_PROJECT',
+      'GOOGLE_CLOUD_PROJECT',
+    },
+  ),
+
+  /// A set of typical environment variables that are likely to represent the
+  /// current Google Cloud instance region.
+  ///
+  /// For context, see:
+  /// * https://cloud.google.com/functions/docs/env-var
+  /// * https://cloud.google.com/compute/docs/gcloud-compute#default_project
+  /// * https://github.com/GoogleContainerTools/gcp-auth-webhook/blob/08136ca171fe5713cc70ef822c911fbd3a1707f5/server.go#L38-L44
```

Comment:
This particular GitHub reference is erroneous from my previous commit; looks like it's carried forward from the project-id definition, and doesn't reference any region related environment variables.